platforms/windows/SQLitePlugin/SQLitePlugin.h

```diff
@@ -0,0 +1,143 @@
+    };
+
+    REACT_STRUCT(ExecuteSqlBatchOptions);
+    struct ExecuteSqlBatchOptions
+    {
+        REACT_FIELD(DBArgs, L"dbargs");
+        DBArgs DBArgs;
+        REACT_FIELD(Executes, L"executes");
+        std::vector<DBQuery> Executes;
+    };
+
+    REACT_STRUCT(OpenOptions);
+    struct OpenOptions
+    {
+        REACT_FIELD(Name, L"name");
+        // Path at which to store the database
+        std::string Name;
+
+        REACT_FIELD(AssetFileName, L"assetFilename");
+        // Optional. When creating the DB, uses this file as the initial state.
+        std::string AssetFileName;
+
+        REACT_FIELD(ReadOnly, L"readOnly");
+        bool ReadOnly;
+    };
+
+    REACT_MODULE(SQLitePlugin, L"SQLite");
+    struct SQLitePlugin : std::enable_shared_from_this<SQLitePlugin>
+    {
+    public:
+        REACT_METHOD(Attach, L"attach");
+        void Attach(
+            JSValue options,
+            std::function<void(std::string)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        REACT_METHOD(Close, L"close");
+        void Close(
+            CloseOptions options,
+            std::function<void(std::string)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        REACT_METHOD(DeleteDB, L"delete");
+        void DeleteDB(
+            DeleteOptions options,
+            std::function<void(std::string)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        REACT_METHOD(EchoStringValue, L"echoStringValue");
+        void EchoStringValue(
+            EchoStringValueOptions options,
+            std::function<void(std::string)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        REACT_METHOD(ExecuteSqlBatch, L"executeSqlBatch");
+        REACT_METHOD(ExecuteSqlBatch, L"backgroundExecuteSqlBatch");
+        void ExecuteSqlBatch(
+            ExecuteSqlBatchOptions options,
+            std::function<void(std::vector<JSValueObject>)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        REACT_METHOD(Open, L"open");
+        void Open(
+            OpenOptions options,
+            std::function<void(std::string)> onSuccess,
+            std::function<void(std::string)> onFailure) noexcept;
+
+        SQLitePlugin::SQLitePlugin();
+        SQLitePlugin::~SQLitePlugin();
+
+    private:
+        std::map<hstring, sqlite3*> OpenDBs;
+        ReactDispatcher SerialReactDispatcher{ ReactDispatcher::CreateSerialDispatcher() };
+
+        static IAsyncOperation<StorageFile> CopyDbAsync(const StorageFile& srcDbFile, const hstring& destDbFileName);
+        static void BindStatement(sqlite3_stmt* stmt_ptr, int argIndex, const JSValue& arg);
+        static bool ExecuteQuery(sqlite3* db, const DBQuery& query, JSValue& result);
+        static JSValue ExtractColumn(sqlite3_stmt* stmtPtr, int columnIndex);
+        static JSValueObject ExtractRow(sqlite3_stmt* stmtPtr);

```

Comment:
nit: don't think we need to indicate types in the parameter name

Reply:
Updated